backend/eventloop/application_event_loop/application_event_runner_deployments.go

```diff
@@ -354,13 +369,13 @@ func (a applicationEventLoopRunner_Action) handleNewGitOpsDeplEvent(ctx context.
 	waitForOperation := !a.testOnlySkipCreateOperation // if it's for a unit test, we don't wait for the operation
 
 	k8sOperation, dbOperation, err := operations.CreateOperation(ctx, waitForOperation, dbOperationInput,
-		clusterUser.Clusteruser_id, operationNamespace, dbQueries, gitopsEngineClient, a.log)
+		clusterUser.Clusteruser_id, engineInstance.Namespace_name, dbQueries, gitopsEngineClient, a.log)
```

Comment:
In the `application_event_runner_*.go` changes, in the functions where you are using gitopsengineinstance (for example, here), can you do a quick sanity check to `engineInstance` after retrieving it, to make sure it's non-nil (and that namspace is non-empty), and if it is, to return an error?

There are a few cases in this file, and also in `application_event_runner_syncrun.go`